### ISSUE TITLE
java.lang.RuntimeException: View is not initialized, height:0 at OverlayView.getVerticalWeight()

### ISSUE THREAD

**Issue by shaubert** [onfido/onfido-android-sdk#10]
Fatal Exception: java.lang.RuntimeException: View is not initialized, height:0
       at com.onfido.android.sdk.capture.ui.camera.OverlayView.getVerticalWeight(Unknown Source)
       at com.onfido.android.sdk.capture.ui.camera.CaptureActivity$7.run(Unknown Source)
       at android.os.Handler.handleCallback(Handler.java:808)
       at android.os.Handler.dispatchMessage(Handler.java:103)
       at android.os.Looper.loop(Looper.java:193)
       at android.app.ActivityThread.main(ActivityThread.java:5333)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:515)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:829)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:645)
       at dalvik.system.NativeStart.main(NativeStart.java)

**Comment by shaubert** [comment]
Devices
HTC 78%
alps 22%

HTC Desire 820G PLUS dual sim 100%

Operating Systems
4 78%
5 22%

**Comment by sergiomarquesmoura** [comment]
Hey @shaubert . From what I understood, you're currently using our `2.3.0` version. A newer `2.4.0` version was released last week, it might be worth to upgrade and check if the issues persist or not.

**Comment by seewah** [comment]
Fixed in version `3.0.0`. See [changelog](https://github.com/onfido/onfido-android-sdk/blob/master/CHANGELOG.MD#300---2018-04-05)

**Comment by compyman** [comment]
I just saw this error come through on an itel S13 running android version 8.10 (onfiod sdk version: 4.4.0)

**Comment by sergiomarquesmoura** [comment]
Hi @compyman . Thanks for flagging this. Could you reproduce it?

**Comment by compyman** [comment]
I can't reproduce it since I don't have access to the devices we are seeing it on
Here's the stack trace I was able to get:
```
java.lang.RuntimeException: View is not initialized, height:0 (Most recent call first)
at com.onfido.android.sdk.capture.ui.camera.OverlayView.getVerticalWeight (Unknown source)
at com.onfido.android.sdk.capture.ui.camera.CaptureActivity$6.run (Unknown source)
at android.os.Handler.handleCallback (Handler.java:790)
at android.os.Handler.dispatchMessage (Handler.java:99)
at android.os.Looper.loop (Looper.java:164)
at android.app.ActivityThread.main (ActivityThread.java:6566)
at java.lang.reflect.Method.invoke (Unknown source)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:438)
at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:857)
```
I have logs but don't necessarily feel good dumping theme here on the public repo - I will send those in with my internal support ticket. 

The phones we've seen this on are:

Infinix HOT 3
BLU STUDIO G2
LG K8 VS500PP

**Comment by compyman** [comment]
there are actually a couple different stack traces: Including here in the case they might be helpful

```
java.lang.RuntimeException: View is not initialized, height:0 (Most recent call first)
at com.onfido.android.sdk.capture.ui.camera.OverlayView.getVerticalWeight (Unknown source)
at com.onfido.android.sdk.capture.ui.camera.CaptureActivity$6.run (Unknown source)
at android.os.Handler.handleCallback (Handler.java:815)
at android.os.Handler.dispatchMessage (Handler.java:104)
at android.os.Looper.loop (Looper.java:194)
at android.app.ActivityThread.main (ActivityThread.java:5637)
at java.lang.reflect.Method.invoke (Unknown source)
at java.lang.reflect.Method.invoke (Method.java:372)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:960)
at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:755)
```

```
java.lang.RuntimeException: View is not initialized, height:0 (Most recent call first)
at com.onfido.android.sdk.capture.ui.camera.OverlayView.getVerticalWeight (Unknown source)
at com.onfido.android.sdk.capture.ui.camera.CaptureActivity$6.run (Unknown source)
at android.os.Handler.handleCallback (Handler.java:815)
at android.os.Handler.dispatchMessage (Handler.java:104)
at android.os.Looper.loop (Looper.java:207)
at android.app.ActivityThread.main (ActivityThread.java:5765)
at java.lang.reflect.Method.invoke (Unknown source)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:789)
at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:679)
```

**Comment by sergiomarquesmoura** [comment]
@compyman @shaubert we found the issue and we're launching a fix for it early next week. I'll update you here once it's live

**Comment by sergiomarquesmoura** [comment]
Hi @shaubert . Thanks for flagging this! 
We just released `4.5.1` which addresses this issue. Please try it out when you have a chance.
